src/NoThunks/Class.hs

```diff
@@ -58,7 +59,9 @@ import Data.Time
 import Data.Void (Void)
 import Data.Word
 import GHC.Stack
+#if !MIN_VERSION_base(4,15,0)
```

Comment:
Can/Should we add a comment what happened in 4.15.0+?

Reply:
I am not even sure why that was needed.

Reply:
@angerman in `4.16.0+` `Natural` is exported from `GHC.TypeLits` which is imported a few lines above this place.  It's possible that the code does not compile with `ghc-9.0`.

Reply:
Indeed it doesn't:
```

src/NoThunks/Class.hs:457:28: error:
    Not in scope: type constructor or class ‘Natural’
    |
457 | deriving via OnlyCheckWhnf Natural instance NoThunks Natural
    |                            ^^^^^^^

src/NoThunks/Class.hs:457:54: error:
    Not in scope: type constructor or class ‘Natural’
    |
457 | deriving via OnlyCheckWhnf Natural instance NoThunks Natural
    |                                                      ^^^^^^^
```